Add floor division and modulo

gustavwilliam opened the pull request:
This adds modulo and floor div to the math module.

gustavwilliam committed:
Add floor division and modulo

decorator-factory commented on gurklang/stdlib_modules/math.py:
```diff
@@ -42,6 +42,26 @@ def add(stack: T[V, T[V, S]], scope: Scope, fail: Fail):
+    if x.tag != "int" or y.tag != "int":
+        fail(f"Cannot perform floor division: {x} and {y} are not both ints")
+    elif y.value == 0:
+        fail(f"Cannot perform floor division: division by zero")
```
I would rather output `Cannot compute {x.value} 0 /` -- this gives more information for debugging. Same in the other function.

decorator-factory commented on gurklang/stdlib_modules/math.py:
```diff
@@ -42,6 +42,26 @@ def add(stack: T[V, T[V, S]], scope: Scope, fail: Fail):
+    if x.tag != "int" or y.tag != "int":
+        fail(f"Cannot get modulo: {x} and {y} are not both ints")
+    elif y.value == 0:
+        fail(f"Cannot perform modulo division: division by zero")
```
I'd use `Cannot compute {x.value} 0 % `

gustavwilliam committed:
Update floor division and modulo error message

Provides the code that failed, with the values. This means including things like "3 0 /" in a division by zero error.

decorator-factory reviewed (approved):
lgtm
